spec/commands/evalsha_spec.rb

```diff
@@ -7,4 +7,14 @@
   it 'returns nothing' do
     @redises.evalsha(script_digest).should be_nil
   end
+
+  context 'with "evalsha" config' do
+    it 'executes the proc' do
+      block = proc { |redis, args| redis.set(args[0], args[1]) }
+      mock = MockRedis.new(evalsha: block)
+
+      mock.evalsha('foo', 'bar')
```

Comment:
This test somewhat misses the point of EVALSHA in that it is supposed to supply a SHA1 digest of the the script. 

While this test is testing _something_, I'm not really clear on the value this provides for you. I'm also not clear on why you are adding an `evalsha` option and using that to determine how to evaluate these.

Can you show a more concrete example of what this enables for you? I'm likely missing something. Appreciate your help.